src/http/gateway/resolver.js

```diff
@@ -0,0 +1,114 @@
+'use strict'
+
+const mh = require('multihashes')
+const promisify = require('promisify-es6')
+const eachOfSeries = require('async/eachOfSeries')
+const debug = require('debug')
+const log = debug('jsipfs:http-gateway:resolver')
+log.error = debug('jsipfs:http-gateway:resolver:error')
+
+const html = require('./utils/html')
+const PathUtil = require('./utils/path')
+
+const INDEX_HTML_FILES = [ 'index.html', 'index.htm', 'index.shtml' ]
+
+const resolveDirectory = promisify((ipfs, path, multihash, callback) => {
+  if (!callback) {
+    callback = noop
+  }
+
+  mh.validate(mh.fromB58String(multihash))
+
+  ipfs
+   .object
+   .get(multihash, { enc: 'base58' })
```

Comment:
I prefer to use the DAG API for this `ipfs.dag.get` if possible (if it isn't, report as a bug why)